### ISSUE TITLE
No objections to GPLv3? 

### ISSUE THREAD

**Issue by pfalcon** [lastgplv2/lastgplv2.org#1]
It's truly amusing that on http://lastgplv2.org/ , the link to objections.html leads to 404.

**Comment by landley** [comment]
On 09/17/2017 07:06 PM, Paul Sokolovsky wrote:
> It's truly amusing that on http://lastgplv2.org/ , the link to
> objections.html leads to 404.

Never finished writing up the page. I switched from using old gcc
toolchains to using llvm and musl-libc and such.

I articulated the fundamental problem I had with GPLv3 years ago:

  https://lkml.org/lkml/2013/5/21/139

And even gave a couple talks on "the rise and fall of copyleft" (which
is a bit hard to follow without the video part of the presentation, but
you can get audio at
https://archive.org/download/OhioLinuxfest2013/24-Rob_Landley-The_Rise_and_Fall_of_Copyleft.mp3
and the outline at http://landley.net/talks/ohio-2013.txt ).

These days GPL has fallen off my radar, I promote public domain
equivalent licenses, namely the "zero clause BSD" I got SPDX approval
for a year or two back. I copied some discussion about that to my blog,
in two parts:

  http://landley.net/notes-2017.html#26-03-2017
  http://landley.net/notes-2017.html#27-03-2017

There's tons of stuff I _could_ do on this front, but I'm busy with
other things these days. I'm happy to let GPL die with the generation of
people who used it. Its penetration in the under-25 crowd appears
negligible, time will take care of the rest.

  http://www.zdnet.com/article/graying-linux-developers-look-for-new-blood/

Rob

**Comment by pfalcon** [comment]
> Never finished writing up the page.

The bottom line is that you might want to put some link in there (or content under existing link). (Or people may think there's a conspiracy of bad guys like you and Linus, who don't value people's freedoms, which GPLv3 is supposed to be protecting better than GPLv1 or GPLv2.)

I appreciate all the links here, but I'm personally aware of the drawbacks of GPLv3 in niche cases very well (though perhaps not in your rendition which is always insightful).

 Oh, btw, coming here on the (echo) of news of toybox having been included in AOSP. Congratulations on achieving this implausible goal! Will it make Android more "self-hosting" and suitable for development? Heh. If Google wanted that, they would have done it like that from the start. It's not on their agenda, the same way as "freedoms" is not on yours or Linus'.

And I find it insightful that you're about the only one of "serious" people who think that Android is useful for something and something should be done about it. Actually, if I have a chance of you listening, something which caught my attention reading your stuff on Android is that you refer to one single "Linux like environment" project in more than one place. I hope you're aware that there're literally dozens (plural) such projects. Literally, every other brave new kid (of those who will type on onscreen keyboard and peer into 5" screens) and their grandma went for a such project. As these kids don't know of cooperation and continuing work of others, they always start from scratch, and of course drop soon, so there's nothing which progresses thru the years. Such project du jour is https://termux.com/, it even appears to come with LLVM. But again, there're dozens, dozens such projects.

**Comment by technosaurus** [comment]
I rolled back a few git histories to the parent of the version switches
https://github.com/technosaurus/gcc-4.2.x-GPL2
https://github.com/technosaurus/binutils-2.17-GPL2
https://github.com/technosaurus/samba3-GPL2
https://github.com/technosaurus/samba4-GPL2
https://github.com/technosaurus/mupdf-GPL2

**Comment by lastgplv2** [comment]
On 01/14/2018 01:16 AM, technosaurus wrote:
> I rolled back a few git histories to the parent of the version switches
> https://github.com/technosaurus/gcc-4.2.x-GPL2
> https://github.com/technosaurus/binutils-2.17-GPL2
> https://github.com/technosaurus/samba3-GPL2
> https://github.com/technosaurus/samba4-GPL2
> https://github.com/technosaurus/mupdf-GPL2

In my old https://landley.net/aboriginal/about.html project I had a
tarball of the last gplv2 commit of binutils, packaged so it actually
built without autoconf installed on the host (just like other release
tarballs). I was trying to do the same for gcc but that last commit
requires gmp and mpfr. The fun part is gcc was in cvs (with a git
mirror), mpfr was in svn, and gmp was in mercurial. I could never figure
out which versions matched up with each other to actually WORK. (And
that's before they metastasized to 5 packages and started rewriting
everything in C++.)
